File: spring-dao/src/main/java/org/r3al/springdao/DaoQueryConfig.java
package org.r3al.springdao;

public interface DaoQueryConfig {

    String getPackageScan();

    default String getFileSufix() {
        return "sql";
    }

    default boolean getUseHibernateTypes() {
        return true;
    }

    default String getSQLDirectory() {
        return DaoQueryAutoConfiguration.SQL_DIRECTORY;
    }

}

File: spring-dao/src/main/kotlin/org/r3al/springdao/DaoQueryConfig.kt
package org.r3al.springdao

interface DaoQueryConfig {
    val packageScan: String?
    val fileSufix: String?
        get() = "xml"
    val useHibernateTypes: Boolean
        get() = true
    val xmlDirectory: String?
        get() = DaoQueryAutoConfiguration.XML_DIRECTORY
}